File: PracticaE02/PracticaE02Back/src/main/java/com/softtek/practicae02back/modelo/Lugar.java
package com.softtek.practicae02back.modelo;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.EqualsAndHashCode;
import lombok.NoArgsConstructor;

@Data
@AllArgsConstructor
@NoArgsConstructor
@EqualsAndHashCode
@Entity
@Table(name = "Lugares")
public class Lugar {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private int id_lugar;
    @Column(length = 60)
    private String nombre;
}

File: PracticaE02/PracticaE02Front/src/app/_model/lugar.ts
export type HTTPLugarResponse = {id_lugar: string, nombre: string};
export class Lugar {

    private _id: string;
    // public get id(): string {
    //     return this._id;
    // }
    // public set id(value: string) {
    //     this._id = value;
    // }

    private _nombre: string;
    public get nombre(): string {
        return this._nombre;
    }
    public set nombre(value: string) {
        this._nombre = value;
    }

    constructor(id: string, nombre: string) {
        this._id = id;
        this._nombre = nombre;
    }

    public static fromHttpResponse(obj: HTTPLugarResponse): Lugar {
        return new Lugar(obj.id_lugar, obj.nombre);
    }
    
}